docs/audio.rst

```diff
@@ -215,6 +244,153 @@ Sound Effects Example
 .. include:: ../examples/soundeffects.py
     :code: python
 
+
+AudioRecording & AudioTrack **V2**
+==================================
+
+There are two classes that can contain (or point to) audio data 
+and an associated sampling rate:
+
+- ``AudioRecording`` contains its own buffer, it's initialised with a size
+  defined in time, and it's the object type that ``microphone.record()``
+  returns.
+- ``AudioTrack`` does not hold its own buffer and instead points to a buffer
+  externally created. This buffer could be an ``AudioRecording``, or a basic
+  type like a ``bytearray``. It's similar to a
+  `memoryview <https://docs.micropython.org/en/v1.9.3/pyboard/reference/speed_python.html#arrays>`_
+  and it can be used to easily chop a portion of the audio data in the
+  ``AudioRecording`` or to modify its contents.
+
+AudioRecording
+--------------
+
+.. py:class::
+    AudioRecording(duration, rate=7812)
+
+    The ``AudioRecording`` object contains audio data and the sampling rate
+    associated to it.
+
+    The size of the internal buffer will depend on the ``rate``
+    (number of samples per second) and ``duration`` parameters.
+
+    :param duration: Indicates how many milliseconds of audio this
+        instance can store.
+    :param rate: The sampling rate at which data will be stored
+        via the microphone, or played via the ``audio.play()`` function.
+
+    .. py:function:: set_rate(sample_rate)
+
+        Configure the sampling rate associated with the data in the
+        ``AudioRecording`` instance.
+
+        :param sample_rate: The sample rate to set.
+
+    .. py:function:: get_rate()
+
+        Return the configured sampling rate for this
+        ``AudioRecording`` instance.
+
+        :return: The configured sample rate.
+
+    .. py:function:: copy()
+
+        :returns: a copy of the ``AudioRecording``.
+
+    .. py:function:: track(start_ms=0, end_ms=-1)
```

Comment:
Are these keyword-only arguments, or positional?

Ie do we allow `recording.track(123)` or require `recording.track(start_ms=123)`?

Reply:
I think it's fine to leave them as positional, users can still use keywords for clarity.

Reply:
I don't think they are positional at the moment:

```python
>>> audio.AudioRecording(100).track(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: extra positional arguments given
```

Reply:
I've now made them positional.